File: kotlinTutorials/kt10/src/main/java/com/java/kt10/LazyNotThreadSaft.java
package com.java.kt10;

/**
 * 基本懒加载
 */
public class LazyNotThreadSaft {
    private static LazyNotThreadSaft instance;

    private LazyNotThreadSaft() {}

    public static LazyNotThreadSaft getInstance() {
        if (null == instance) {
            instance = new LazyNotThreadSaft();
        }
        return instance;
    }
}

File: kotlinTutorials/kt10/src/main/kotlin/com/kotlin/kt10/LazyNotThreadSaft.kt
package com.kotlin.kt10

/**
 * 基本懒加载
 */
class LazyNotThreadSaft {
    companion object {
        val instace by lazy (LazyThreadSafetyMode.NONE) {
            LazyNotThreadSaft()
        }
        var instance2: LazyNotThreadSaft? = null

        fun getInstance(): LazyNotThreadSaft {
            if (instance2 == null) {
                instance2 = LazyNotThreadSaft()
            }
            return instance2!!
        }
    }
}